samplers/aws/xray/remote_sampler.go

```diff
@@ -0,0 +1,221 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xray
+
+import (
+	"context"
+	crypto "crypto/rand"
+	"errors"
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/otel/sdk/trace"
+)
+
+// RemoteSampler is an implementation of SamplingStrategy.
+type RemoteSampler struct {
+	// List of known centralized sampling rules
+	manifest *centralizedManifest
+
+	// proxy is used for getting quotas and sampling rules
+	xrayClient *xrayClient
+
+	// pollerStart, if true represents rule and target pollers are started
+	pollerStart bool
+
+	// Unique ID used by XRay service to identify this client
+	clientID string
+
+	// Provides system time
+	clock Clock
+
+	mu sync.RWMutex
+}
+
+// Compile time assertion that remoteSampler implements the Sampler interface.
+var _ trace.Sampler = (*RemoteSampler)(nil)
+
+// NewRemoteSampler returns a centralizedSampler which decides to sample a given request or not.
+func NewRemoteSampler() *RemoteSampler {
+	// Generate clientID
+	var r [12]byte
+
+	_, err := crypto.Read(r[:])
+	if err != nil {
+		log.Println("error reading cryptographically secure random number generator")
+		return nil
```

Comment:
```suggestion
		return fmt.Errorf("unable to generate client ID: %w", err)
```

Libraries generally shouldn't log errors, but return them to the caller who can decide what to do with them.